### ISSUE TITLE
[BUG] Can't pip install directly from SCM into onedir environment

### ISSUE THREAD

**Issue by moloney** [saltstack/salt#64911]
**Description**

Trying to use "pip.install" to install from source control (e.g. github) always fails on a onedir install, apparently due to a failure to locate the "setuptools_scm" module, even though it appears to be installed in the onedir environment.

**Setup**

Just need a onedir install of 3005.1.

**Steps to Reproduce the behavior**

```
$ salt myminion pip.install git+https://github.com/moloney/sh.git
myminion:
    ----------
    pid:
        119355
    retcode:
        1
    stderr:
          Running command git clone --filter=blob:none --quiet https://github.com/moloney/sh.git /tmp/pip-req-build-xya5rsni
          error: subprocess-exited-with-error
          
          python setup.py egg_info did not run successfully.
          exit code: 1
          
          [25 lines of output]
          Traceback (most recent call last):
            File "tiamatpip/cli.py", line 383, in _handle_runpy_return
              runpy_f(*args, **kwargs)
            File "runpy.py", line 87, in _run_code
            File "<tiamat-pip-run-code>", line 2, in <module>
            File "<pip-setuptools-caller>", line 34, in <module>
            File "/tmp/pip-req-build-xya5rsni/setup.py", line 21, in <module>
              setup(
            File "setuptools/__init__.py", line 155, in setup
              return distutils.core.setup(**attrs)
            File "distutils/core.py", line 108, in setup
              _setup_distribution = dist = klass(attrs)
            File "setuptools/dist.py", line 454, in __init__
              _Distribution.__init__(
            File "distutils/dist.py", line 292, in __init__
              self.finalize_options()
            File "setuptools/dist.py", line 844, in finalize_options
              for ep in sorted(loaded, key=by_order):
            File "setuptools/dist.py", line 843, in <lambda>
              loaded = map(lambda e: e.load(), filtered)
            File "pkg_resources/__init__.py", line 2480, in load
              return self.resolve()
            File "pkg_resources/__init__.py", line 2486, in resolve
              module = __import__(self.module_name, fromlist=['__name__'], level=0)
          ModuleNotFoundError: No module named 'setuptools_scm'
          [end of output]
          
          note: This error originates from a subprocess, and is likely not a problem with pip.
        error: metadata-generation-failed
        
        Encountered error while generating package metadata.
        
        See above for output.
        
        note: This is an issue with the package mentioned above, not pip.
        hint: See above for details.
    stdout:
        Collecting git+https://github.com/moloney/sh.git
          Cloning https://github.com/moloney/sh.git to /tmp/pip-req-build-xya5rsni
          Resolved https://github.com/moloney/sh.git to commit 0bb90938ad3f4a9f7c551087fc8475e00c830275
          Preparing metadata (setup.py): started
          Preparing metadata (setup.py): finished with status 'error'
```

The setuptools-scm package is installed in the onedir env:

```
$ salt-pip freeze | grep setuptools-scm
setuptools-scm==7.1.0
setuptools-scm-git-archive==1.4
```

**Expected behavior**
Should be possible to install directly from github.

**Versions Report**

Salt Version:
          Salt: 3005.1
 
Dependency Versions:
          cffi: 1.14.6
      cherrypy: unknown
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.1.0
       libgit2: Not Installed
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: 2.21
      pycrypto: Not Installed
  pycryptodome: 3.9.8
        pygit2: Not Installed
        Python: 3.9.16 (main, Jan  6 2023, 22:50:07)
  python-gnupg: 0.4.8
        PyYAML: 5.4.1
         PyZMQ: 23.2.0
         smmap: Not Installed
       timelib: 0.2.4
       Tornado: 4.5.3
           ZMQ: 4.3.4
 
System Versions:
          dist: ubuntu 22.04 jammy
        locale: utf-8
       machine: x86_64
       release: 5.15.0-56-generic
        system: Linux
       version: Ubuntu 22.04 jammy

**Comment by moloney** [comment]
This does at least seem to be fixed in 3006.1.

**Comment by garethgreenaway** [comment]
@moloney Thanks for the report.  Great to hear that this particular bug is fixed in 3006.1, since the 3005 release will only be receiving security fixes going forward our recommendation is to upgrade to the latest 3006 release.
